### ISSUE TITLE
IRIS: Support for Multiple Eurotherms

### ISSUE THREAD

**Issue by kjwoodsISIS** [ISISComputingGroup/IBEX#1403]
As scientist working on IRIS, I need to monitor and control the temperature from up to 6 Eurotherm devices.

##### Acceptance Criteria
1. I can view and change the PVs which define the state of multiple Eurotherm Controllers.
1. I can view and change the state of multiple Eurotherm Controllers via the IBEX GUI.

##### Notes
1. May need to modify the existing Eurotherm IOC & OPI.
1. IRIS will need up to 6 Eurotherm controllers.  We should make the number of supported Eurotherm controllers configurable.
   1. One of the Eurotherm controllers is used to monitor the temperature of the CCR
   1. The other 5 Eurotherm controllers are used to monitor the temperature of the samples in the sample can.
   1. Need to have an IOC that can control N Eurotherms on a single COM port ( N might be as large as 6 or 7).
   1. May need to have multiple IOCs - to control multiple groups of Eurotherms on different COM ports.

**Comment by John-Holt-Tessella** [comment]
Eurotherm appears to be broken, it crashes on startup. last lines before death are
```
## For example: eurotherm address 10 => GAD = 1 and LAD = 0
dbLoadRecords("C:/Instrument/Apps/EPICS/ioc/master/EUROTHERM/db/devEurotherm.db"
,"P=NDW1407:HGV27692:EUROTHERM1:A01:, Q=NDW1407:HGV27692:EUROTHERM1:, GAD=0, LAD
=1, PORT=L0, LDIR = C:/Instrument/Settings, SDIR=C:/Instrument/Settings/calib/se
nsors, READ=READASCII1, RAMPLIST=RAMPFILELIST1, RAMPSEARCH=.*, SENSORLIST=SENSOR
FILELIST1, SENSORSEARCH=^C.*")
Can't set "NDW1407:HGV27692:EUROTHERM1:A01:RAMP_DIRBASE:SP.VAL" to "C:/Instrumen
t/Settings"
Error at or before ")" in file "C:/Instrument/Apps/EPICS/ioc/master/EUROTHERM/db
/devEurotherm.db" line 8
Can't set "NDW1407:HGV27692:EUROTHERM1:A01:RAMP_PATTERN.VAL" to ".*"
Error at or before ")" in file "C:/Instrument/Apps/EPICS/ioc/master/EUROTHERM/db
/devEurotherm.db" line 49
Can't set "NDW1407:HGV27692:EUROTHERM1:A01:READ_RAMP_DIRBASE:SP.VAL" to "C:/Inst
rument/Settings"
Error at or before ")" in file "C:/Instrument/Apps/EPICS/ioc/master/EUROTHERM/db
/devEurotherm.db" line 61
Record "NDW1407:HGV27692:EUROTHERM1:A01:READ_RAMP_DIRBASE:SP" does not have a fi
eld "DOL"
Error at or before ")" in file "C:/Instrument/Apps/EPICS/ioc/master/EUROTHERM/db
/devEurotherm.db" line 62
Record "NDW1407:HGV27692:EUROTHERM1:A01:READ_RAMP_DIRBASE:SP" does not have a fi
eld "OMSL"
Error at or before ")" in file "C:/Instrument/Apps/EPICS/ioc/master/EUROTHERM/db
/devEurotherm.db" line 65
```

**Comment by KathrynBaker** [comment]
Having tried to run it on my system, I don’t even get as far as any of the dbLoadRecords commands, so the problem may be more fundamental.

**Comment by John-Holt-Tessella** [comment]
Garether has provided me with a way to connect to the IRIS one if it is not me that pick up this ticket let me know and we can try making it work on your machine.

**Comment by John-Holt-Tessella** [comment]
The reason for the crash is that the sseq record is defined differently in std and calc because of version conflicts. @FreddieAkeroyd suggested updating the version of std so that it is compatible using a vendor branch which I will now do.

**Comment by AdrianPotter** [comment]
https://github.com/ISISComputingGroup/ibex_gui/pull/296

**Comment by AdrianPotter** [comment]
A few findings from testing:

- If I start the IOC without a calibration file, then the calibration directory will be in alarm state. Once I set a calibration file, I'd expect that to disappear but it doesn't.
- When I set the ramp dir in the OPI I get an error `Failed to write value to PV NDW1695:HRA63823:EUROTHERM6:A02:READ_RAMP_DIRBASE:SP...`
- When the IOC is placed in RecSim mode from the GUI, many of the PVs remain disconnected. When placed in sim mode properly (i.e. via `caget`) it works as expected.
- When a file is entered as the sensor calibration file in the GUI, is it possible to issue an alarm if the file does not exist?
